optuna/study/_optimize.py

```diff
@@ -237,93 +234,39 @@ def _run_trial(
 
     # `Study.tell` may raise during trial post-processing.
     try:
-        study.tell(trial, values=values, state=state)
-    except Exception:
+        frozen_trial = study.tell(trial, values=value_or_values, state=state)
+    except Exception as e:
+        tell_err = e
         raise
     finally:
-        if state == TrialState.COMPLETE:
-            study._log_completed_trial(trial, cast(List[float], values))
-        elif state == TrialState.PRUNED:
-            _logger.info("Trial {} pruned. {}".format(trial.number, str(func_err)))
-        elif state == TrialState.FAIL:
+        # Even if an error happens in Study.tell, this logging section
+        # should be reached. Because `frozen_trial` would not be assigned when
+        # Study.tell fails, it fetches the trial from the storage.
+        if tell_err is not None:
+            frozen_trial = study._storage.get_trial(trial._trial_id)
```

Comment:
Not a strong opinion but would the code become cleaner/no need for descriptive comment if we do this trial query inside the except clause/indent?